### PR TITLE
No output toggle for az login within the AzureCLIV2 task 

### PR DESCRIPTION
**Pull request by piotr-gbyliczek**
**Task name**: AzureCLIV2

**Description**: Added a toggleable setting for suppressing output from az login

**Documentation changes required:** (Y/N) Yes

**Added unit tests:** (Y/N) No

**Attached related issue:** (Y/N) #14156

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

**Review by syedhamjath — APPROVED** [review]

File: Tasks/AzureCLIV2/task.json [hunk 1]
```diff
@@ -171,7 +171,16 @@
...
             "groupName": "advanced"
-        }
+        },
+        {
+            "name": "visibleAzLogin",
+            "type": "boolean",
+            "label": "az login output visibility",
+            "defaultValue": "true",
+            "required": false,
+            "helpMarkDown": "If this is set to true, az login command will output to the task. Setting it to true will suppress the az login output",
```
Comment (syedhamjath) [inline Tasks/AzureCLIV2/task.json:181]: Please rephrase this help text and explain what will happen when it is set to false.

**Review by syedhamjath — CHANGES_REQUESTED** [review]

File: Tasks/AzureCLIV2/task.json (hunk 1 above)
Comment (piotr-gbyliczek) [reply Tasks/AzureCLIV2/task.json]: Changed the parameter description.

**Review by syedhamjath — APPROVED** [review]

**Review by syedhamjath — APPROVED** [review]

**Comment by piotr-gbyliczek** [comment]
How do I merge this PR? It seems to be in constant wait for some checks or get's outdated and update resets the approvals.

**Comment by syedhamjath** [comment]
@issacnitin Can someone help us on this pull request?

**Review by syedhamjath — APPROVED** [review]

**Comment by piotr-gbyliczek** [comment]
@syedhamjath Thanks for the approval again, but I am still not able to complete the PR on my side. In fact I would expect this to be done on your side as the change would be merged into the repo you are owning.

**Comment by piotr-gbyliczek** [comment]
Thanks for approving it again @syedhamjath, but I'm still stuck with this not giving me the option to merge the change. 
It looks like the issue is with the checks, which I have no idea how to complete - I was expecting them to be done as part of the build, which is triggered automatically after any update.

**Review by mieel — APPROVED** [review]

**Comment by syedhamjath** [comment]
@piotr-gbyliczek I don't think I have write permission to the repo.

**Review by syedhamjath — APPROVED** [review]

**Review by syedhamjath — APPROVED** [review]

**Comment by DanielHRio** [comment]
@syedhamjath I reckon this would be useful, are you able to update and merge it? Thank you.

**Comment by syedhamjath** [comment]
@DanielHRio No, we are struggling with this pull request and review process.

**Comment by piotr-gbyliczek** [comment]
@DanielHRio I can't seem to get a merge here due to these statuses for all CI pipelines. I do not understand what needs to be done with them, and if this is something I need to do, or some sort of automation. Feel free to advise if you can.

**Comment by piotr-gbyliczek** [comment]
@v-ibshaik @bishal-pdMSFT Can one of you help with this PR (as per comment in #15845 you may be able to)?

**Comment by piotr-gbyliczek** [comment]
@microsoft-github-policy-service agree

**Comment by piotr-gbyliczek** [comment]
Hi @max-zaytsev,
I can see you tried to run pipelines on this pull request, but it hit some conflicts in merge. I have resolved that now, so could you please retry with the pipelines? 

Also, if there is anything else I need (or needed) to do to complete this PR, please advise on what it would be - I'm not regular commiter, so not sure about all procedures.

**Comment by piotr-gbyliczek** [comment]
Bumped the version of task to current sprint.

**Comment by LeftTwixWand** [comment]
You need to bump the task's version

**Comment by ransagy** [comment]
@LeftTwixWand Any chance you can help push this along?

**Comment by geekzter** [comment]
@ransagy you can control Azure CLI (logging) behavior using [environment variables](https://learn.microsoft.com/cli/azure/azure-cli-configuration#cli-configuration-values-and-environment-variables):
```yaml
  env:
    AZURE_CORE_OUTPUT: none
```
Then override the env variable or each command individually in your script.

**Comment by ransagy** [comment]
@geekzter Thanks for that reference; while being somewhat of a workaround, it almost works - with your suggestion and even with the extra only-show-errors flag, the `az --version` default call still outputs to the log, adding at least 18 lines to any CLI task. the other commands do seem suppressed.

**Comment by LeftTwixWand** [comment]
Hey @ransagy let me discuss it with my manager and prioritize it. I think, we would be able to merge it this week an release next week.

**Comment by piotr-gbyliczek** [comment]
@LeftTwixWand Any way to move this PR forward would be appreciated.

I wrote this fix about 2 years ago and sure lost visibility of it for months due to no response/advice, then was assigned to another project which does not use Azure, so no longer directly interested, but there seems to be need for this and my former project could finally get rid of the workaround code (using env variable method described above) and gain some visibility and clarity in that section of the pipeline :)

I keep bumping minor and patch versions every now and then as I don't want to abandon the work, but over 2 years it has been random.

**Comment by piotr-gbyliczek** [comment]
It seems to be back to being ignored, any luck discussing priority on this @LeftTwixWand ?

**Comment by LeftTwixWand** [comment]
Hey @piotr-gbyliczek I'll try to review and test this pr today. I think, we're good to go with this changes

**Review by LeftTwixWand — APPROVED** [review]

**Comment by LeftTwixWand** [comment]
Let's wait for the pipelines to be complete

**Comment by piotr-gbyliczek** [comment]
Great, thanks @LeftTwixWand

**Comment by LeftTwixWand** [comment]
@grzmax I'll take a look

**Pull request merged**